crates/sui-core/src/authority.rs

```diff
@@ -849,12 +848,44 @@ impl AuthorityState {
                     );
                     tx_guard.release();
                 } else {
-                    error!(?digest, "commit_certificate failed: {}", err);
+                    error!(?digest, seq=?ticket_seq, "commit_certificate failed: {}", err);
+                }
+                debug!(
+                     seq=?ticket_seq,
+                    "Ticket not notified due to commit failure",
+                );
+                // Check if we were able to sequence the tx at all
+                if let Some(tx_seq) = self.db().get_tx_sequence(*certificate.digest()).await? {
```

Comment:
I am probably being picky, but this line could return Err as well..

Reply:
I am ok with this going in though, as the chance should be really really really low. We need to start somewhere and this is a big improvement already

Reply:
Yeah I agree, i thought of it too and wanted to add a few retries there to handle db read failing. Let me do it in a follow up PR, and we can let this go for now